tests/e2e/testutils.go

```diff
@@ -306,13 +306,10 @@ func KillK3sCluster(nodes []VagrantNode) error {
 		if _, err := node.RunCmdOnNode("k3s-killall.sh"); err != nil {
 			return err
 		}
-		if _, err := node.RunCmdOnNode("sh -c 'docker ps -qa | xargs -r docker rm -fv'"); err != nil {
-			return err
-		}
 		if _, err := node.RunCmdOnNode("rm -rf /etc/rancher/k3s/config.yaml.d /var/lib/kubelet/pods /var/lib/rancher/k3s/agent/etc /var/lib/rancher/k3s/agent/containerd /var/lib/rancher/k3s/server/db /var/log/pods /run/k3s /run/flannel"); err != nil {
 			return err
 		}
-		if _, err := node.RunCmdOnNode("systemctl restart containerd docker"); err != nil {
+		if _, err := node.RunCmdOnNode("systemctl restart containerd"); err != nil {
```

Comment:
you can remove the containerd restart from here as well - the host containerd is the backend for host docker, and is otherwise unused in favor of the bundled containerd.